overlord/snapstate/handlers.go

```diff
@@ -880,6 +883,9 @@ func (m *SnapManager) doPreDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 		refreshInfo := busyErr.PendingSnapRefreshInfo()
 		return asyncRefreshOnSnapClose(m.state, refreshInfo)
 	}
+	// TODO: in the future keep the lock until the refresh finishes so there's no
+	// change of the snap starting between changes and preventing the auto-refresh
```

Comment:
this is the full snap lock, I don't think we ever keep it for long? so not sure the comment applies